File: java/Two/Two/Plan.java
public abstract class Plan {
    private float price;
    private String name;

    public Plan(String name, float price) {
        this.name = name;
        this.price = price;
    }

    public abstract String[] getServices();
    public abstract float calculateCost();

    public float getPrice() {
        return price;
    }

    public String getName() {
        return name;
    }
}

File: Two/plan.php
<?php

abstract class Plan {
    private float $price;
    private string $name;

    public function __construct(string $name, float $price) {
        $this->name = $name;
        $this->price = $price;
    }

    abstract public function getServices(): array;
    abstract public function calculateCost(): float;

    public function getPrice(): float {
        return $this->price;
    }

    public function getName(): string {
        return $this->name;
    }
}

?>